src/lib/slider/slider.ts

```diff
@@ -265,14 +266,28 @@ export class MdSlider implements AfterContentInit, ControlValueAccessor {
    */
   snapThumbToValue() {
     this.updatePercentFromValue();
-    this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
+    if (this._sliderDimensions) {
+      this._renderer.updateThumbAndFillPosition(this._percent, this._sliderDimensions.width);
+    }
+  }
+
+  /**
+   * Initializes the tick and slider positions.
+   * @private
+   */
+  private _initTicksAndSlider() {
```

Comment:
In this case I think it would be better to call the individual functions in the setters / init because it's more clear in those locations what's happening; you won't know what `_initTicksAndSlider` does precisely without jumping to the definition.